criar e editar albuns

LaraaMagalhaes opened the pull request:
arrumei o searchview, da pra criar albun, da pra editar o nome e compartilhar os albuns, da pra salvar artes nos albuns, acho q so.

LaraaMagalhaes committed:
criar e editar albuns

arrumei o searchview, da pra criar albun, da pra editar o nome e compartilhar os albuns, da pra salvar artes nos albuns, acho q so.

Joao-ava commented on app/src/main/java/com/example/appcultural/data/FirebaseAlbumsRepository.kt:
```diff
@@ -0,0 +1,45 @@
+    private val db = FirebaseFirestore.getInstance()
+    private val collection = db.collection("albums")
+
+    suspend fun listAlbums(): List<Album> {
```
Modificar nomes das funções para remover o nome da entidade, como na classe **FirebaseAlbumsRepository** já está explicado que é sobre Álbum

Joao-ava commented on app/src/main/java/com/example/appcultural/data/FirebaseArtsRepository.kt:
```diff
@@ -1,33 +1,37 @@
     suspend fun add(art: Art): Art {
         val result = collection.add(art).await()
-        art.id = result.id
-        result.set(art).await()
```
voltar a fazer `result.set(art).await()` pois isso faz atualizar o campo id interno do documento

Joao-ava commented on app/src/main/java/com/example/appcultural/entities/Art.kt:
```diff
@@ -9,5 +9,6 @@ data class Art(
     public val genders: List<String> = listOf(),
-    public val location: ArtLocation = ArtLocation(10, 10)
+    public val location: ArtLocation = ArtLocation(10, 10),
+    public val albumId: String = ""
```
A entidade de obra de arte não faz sentido ter `albumId` já que uma obra de arte pode estar em vários álbuns, faz mais sentido ter a referencia da obra de arte no álbum como é feito com o campo `artIds` em `Album.kt`.

LaraaMagalhaes committed:
nome das funçoes,  result.set(art).await() e albumid

voltei com o result.set(art).await(), mudei o nome das funçoes e tirei o albumId do  art.kt
